public/model/epi-data-store.js

```diff
@@ -0,0 +1,143 @@
+        {start_time: utc_date(2016, 2, 1), end_time: utc_date(2016, 2, 8),
+         region_case_data: {4589: {fake_dengue: 111},
+                            4611: {fake_dengue: 121, fake_zika: 122},
+                            4877: {fake_dengue: 1131, fake_zika: 1132}}
+        },
+        // Feb 8 - Feb 15: Missing dengue data for 4611. Also, 4877 has chikungunya data.
+        {start_time: utc_date(2016, 2, 8), end_time: utc_date(2016, 2, 15),
+         region_case_data: {4589: {fake_dengue: 211, fake_zika: 212},
+                            4611: {fake_zika: 222},
+                            4877: {fake_dengue: 1231, fake_zika: 1232, fake_chikungunya: 1233}}
+        },
+        // Feb 15 - Feb 29: Only has data for 4611. 2 weeks.
+        {start_time: utc_date(2016, 2, 15), end_time: utc_date(2016, 2, 29),
+         region_case_data: {4611: {fake_dengue: 321, fake_zika: 322}}
+        },
+        // Feb 15 - Mar 8: Only has data fro 4589 and 4877. Overlaps with previous time span,
+        // but longer.
+        {start_time: utc_date(2016, 2, 15), end_time: utc_date(2016, 3, 8),
+         region_case_data: {4589: {fake_dengue: 311, fake_zika: 312},
+                            4877: {fake_dengue: 1331, fake_zika: 1332}}
+        }
+      ];
+      this.most_recent_start_time = _.last(this.data_by_date_interval_and_region).start_time;
+    }).bind(this));
+  },
+
+  // Return a recent epi data record. If there are multiple records that include
+  // the given `date`, returns the one that has data on the most regions. If
+  // there are no records that include `date`, returns the most recent record
+  // that occurred before the date.
+  get_best_recent_epi_data: function(date) {
+    // TODO(jetpack): binary search instead of linear scan.
+    var matching_records = _.filter(this.data_by_date_interval_and_region, function(record) {
+      return record.start_time <= date && date < record.end_time;
+    });
+    if (matching_records.length) {
+      console.log('EDS GBRED matching records:', matching_records);
+      var best_record = _.reduce(matching_records, function(best_record, record) {
+        return (_.size(record.region_case_data) > _.size(best_record.region_case_data)) ?
+          record : best_record;
+      });
+      return best_record;
+    }
+    // No epi records contain `date`. Find the record with the most recent
+    // `start_time` that occurred before the date.
+    // TODO(jetpack): binary search instead of linear scan.
+    console.log('EpiDataStore doesnt have matching data for', date,
+                ', just looking for most recent data now..');
+    return _.findLast(this.data_by_date_interval_and_region, function(record) {
+      console.log('testing if', record.start_time, 'older than', date, 'for record', record);
+      return record.start_time < date;
+    });
+  },
+
+  get_recent_epi_data_for_region: function(date, region_code) {
+    // TODO(jetpack): binary search instead of linear scan.
```

Comment:
This will probably hold for quite a while, since data will almost certainly be less frequent than 1/week. But good point.